music21/expressions.py

```diff
@@ -1190,12 +1716,16 @@ def realize(self, srcObj: 'music21.note.Note', *, inPlace=False):
         music21.expressions.ExpressionException: The note is not long enough to realize a turn
         '''
         from music21 import key
+
+        if not hasattr(srcObj, 'pitch'):
+            raise TypeError('Cannot realize turn with Unpitched srcObj')
```

Comment:
we can change everything to `if not srcObj.pitches` and handles Rests, etc. -- but the typing is set to music21.note.Note so it's an error if any routine above is calling realize on anything that is not a Note object.

Reply:
Ah yes, this is where I saw the Trill on an Unpitched (snare drum), which crashed until I added this code.  I'll fix the typing to be GeneralNote, and use `if not srcObj.pitches` to defend.

Reply:
Done.